### PR TITLE
Add config hash module

### PR DESCRIPTION
**Pull request by Siegrift**
Closes https://github.com/api3dao/commons/issues/48

## Rationale

Explained in https://github.com/api3dao/signed-api/issues/240. I was thinking whether to include also the utilities to sign the message, but opted not to (because signing the hash is not required - e.g. Airseeker will likely not do it). So the module is basically about uniquely serialising plain JS object (a.k.a config) and producing a `sha256` hash. I also verified that the same hash can be computed using a combination of UNIX utilities.

### CODE REVIEW

File: src/config-hash/README.md
Comment (andreogle) [inline src/config-hash/README.md:1]: I wonder if adding links to all of these nested READMEs in the main README would be nice. Provides a bit of a nicer DX than having to browse through the `src/` folder

File: src/config-hash/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { createHash } from 'node:crypto';
+
+import { isObject } from 'lodash';
```
Comment (andreogle) [inline src/config-hash/index.ts:3]: Same comment as https://github.com/api3dao/commons/pull/54#discussion_r1505004128

```suggestion
import isObject from 'lodash/isObject';
```

File: test/fixtures/config-hash/file.json
Comment (andreogle) [inline test/fixtures/config-hash/file.json:1]: maybe worth inlining this too?

**Review by andreogle — APPROVED** [review]

File: test/fixtures/config-hash/file.json
Comment (Siegrift) [reply test/fixtures/config-hash/file.json:1]: I want to read this via UNIX commands so [here](https://github.com/api3dao/commons/pull/51/files#diff-75acae6fc473fe10d20919c0812287092c72ba4b0aae22b99cd38d64d945c01cR89) so I need to keep it as a file.

File: src/config-hash/README.md
Comment (Siegrift) [reply src/config-hash/README.md:1]: Added. I've also added a small unit test that ensures they are up to date with implementation.

Related to that, I wonder how much value do the READMEs provide. As of now the modules are quite small, so the README feels like a "formality". WDYT ?

**Pull request merged**